Pod/Classes/WPInputMediaPickerViewController.m

```diff
@@ -68,19 +69,75 @@ - (void)setupMediaPickerViewController {
+    [self configureCollectionView];
+}
 
-    CGFloat spacing = 1.0f;
-    CGFloat size = floorf((self.view.frame.size.height - spacing) / 2.0);
-    self.mediaPicker.options.cameraPreviewSize = CGSizeMake(1.5*size, 1.5*size);
+- (void)configureCollectionView {
+    CGFloat photoSpacing = 1.0f;
+    CGFloat photoSize;
     UICollectionViewFlowLayout *layout = [[UICollectionViewFlowLayout alloc] init];
-    layout.scrollDirection = UICollectionViewScrollDirectionHorizontal;
-    layout.itemSize = CGSizeMake(size, size);
-    layout.minimumLineSpacing = spacing;
-    layout.minimumInteritemSpacing = spacing;
-    layout.sectionInset = UIEdgeInsetsMake(0, 5, 0, 10);
 
+    if (self.scrollVertically) {
+        CGFloat frameWidth = self.view.frame.size.width;
+        NSUInteger numberOfPhotosForLine = [self numberOfPhotosPerRow:frameWidth];
+
+        photoSize = [self.mediaPicker cellSizeForPhotosPerLineCount:numberOfPhotosForLine
+                                                       photoSpacing:photoSpacing
+                                                         frameWidth:frameWidth];
+
+        // Check the actual width of the content based on the computed cell size
+        // How many photos are we actually fitting per line?
+        CGFloat totalSpacing = (numberOfPhotosForLine - 1) * photoSpacing;
+        numberOfPhotosForLine = floorf((frameWidth - totalSpacing) / photoSize);
+
+        CGFloat contentWidth = (numberOfPhotosForLine * photoSize) + totalSpacing;
+
+        // If we have gaps in our layout, adjust to fit
+        if (contentWidth < frameWidth) {
+            photoSize = [self.mediaPicker cellSizeForPhotosPerLineCount:numberOfPhotosForLine
+                                                           photoSpacing:photoSpacing
+                                                             frameWidth:frameWidth];
+        }
+
+        layout.scrollDirection = UICollectionViewScrollDirectionVertical;
+        layout.sectionInset = UIEdgeInsetsMake(2, 0, 0, 0);
+    } else {
+        photoSize = floorf((self.view.frame.size.height - photoSpacing) / 2.0);
+        layout.scrollDirection = UICollectionViewScrollDirectionHorizontal;
+        layout.sectionInset = UIEdgeInsetsMake(0, 5, 0, 5);
+    }
+
+    layout.itemSize = CGSizeMake(photoSize, photoSize);
+    layout.minimumLineSpacing = photoSpacing;
+    layout.minimumInteritemSpacing = photoSpacing;
+    self.mediaPicker.options.cameraPreviewSize = CGSizeMake(1.5*photoSize, 1.5*photoSize);
     self.mediaPicker.collectionView.collectionViewLayout = layout;
+}
 
+/**
+ Given the provided frame width, this method returns a progressively increasing number of photos 
+ to be used in a picker row.
+ 
+ @param frameWidth Width of the frame containing the picker
+
+ @return The number of photo cells to be used in a row. Defaults to 3.
+ */
+- (NSUInteger)numberOfPhotosPerRow:(CGFloat)frameWidth {
+    NSUInteger numberOfPhotos = 3;
+
+    if (frameWidth >= 375 && frameWidth < 500) {
```

Comment:
The only request I have since this is the best way to do this is to convert these numbers to constants please so that they're not just magic numbers to someone jumping in here.